src/MySqlConnector/MySqlClient/ConnectionPool.cs

```diff
@@ -138,6 +147,14 @@ public static ConnectionPool GetPool(ConnectionSettings cs)
 				await pool.ClearAsync(ioBehavior, cancellationToken).ConfigureAwait(false);
 		}
 
+		public static async Task ReapPoolsAsync(IOBehavior ioBehavior, CancellationToken cancellationToken)
+		{
+			var pools = new List<ConnectionPool>(s_pools.Values);
```

Comment:
My understanding is that the `Values` collection is thread-safe and can be enumerated on one thread while the dictionary is being modified on a different thread. (It will represent a snapshot of the dictionary at the time the property was accessed.)

Thus, the local copy of `s_pools.Values` is unnecessary and this could be inlined. (Same for the existing `ClearPoolsAsync`.)